addons/binding/org.openhab.binding.loxone/src/main/java/org/openhab/binding/loxone/internal/core/LxWsClient.java

```diff
@@ -644,141 +670,289 @@ public void onMessage(String msg) {
+                            setClientState(ClientState.RUNNING);
+                            notifyMaster(EventType.SERVER_ONLINE, null, null);
+                            if (sendCmdWithResp(CMD_ENABLE_UPDATES, false, false) == null) {
+                                notifyAndClose(LxOfflineReason.COMMUNICATION_ERROR, "Failed to enable state updates.");
+                            }
+                        } catch (JsonParseException e) {
+                            notifyAndClose(LxOfflineReason.INTERNAL_ERROR, "Error processing received configuration");
+                        }
+                        break;
+                    case CLOSING:
+                    default:
+                        break;
                 }
             }
         }
 
+        LxJsonSubResponse getSubResponse(String msg) {
+            try {
+                LxJsonSubResponse subResp = gson.fromJson(msg, LxJsonResponse.class).subResponse;
+                if (subResp == null) {
+                    logger.debug("[{}] Miniserver response subresponse is null: {}", debugId, msg);
+                    return null;
+                }
+                if (subResp.control == null) {
+                    logger.debug("[{}] Miniserver response control is null: {}", debugId, msg);
+                    return null;
+                }
+                return subResp;
+            } catch (JsonSyntaxException e) {
+                logger.debug("[{}] Miniserver response JSON parsing error: {}, {}", debugId, msg, e.getMessage());
+                return null;
+            }
+        }
+
         /**
-         * Stops keep alive thread and ceases sending keep alive messages to the Miniserver
+         * Returns {@link Gson} object so it can be reused without creating a new instance.
+         *
+         * @return
+         *         Gson object for reuse
          */
-        private void stopKeepAlive() {
-            logger.trace("[{}] stopping keepalives in state {}", debugId, state);
-            if (keepAlive != null) {
-                keepAlive.cancel(true);
-                keepAlive = null;
+        Gson getGson() {
+            return LxWsClient.this.getGson();
+        }
+
+        /**
+         * Send a HTTP GET request and return server's response.
+         *
+         * @param request
+         *            request content
+         * @return
+         *         response received
+         */
+        String httpGet(String request) {
+            try {
+                URL url = new URL("http", host.getHostAddress(), port,
+                        request.startsWith("/") ? request : "/" + request);
+                HttpURLConnection con = (HttpURLConnection) url.openConnection();
+                con.setRequestMethod("GET");
+                StringBuilder result = new StringBuilder();
+                BufferedReader reader = new BufferedReader(new InputStreamReader(con.getInputStream()));
```

Comment:
And put these streams in an try block so the will be closed automatically.

Reply:
Good catch ;).  You should be able to use try with resource to auto close the streams.